### PR TITLE
Export the script we use to make release source tarballs

### PR DESCRIPTION
**Pull request by morlovich**
(this was tweaked slightly after the 34 release)

### CODE REVIEW

File: devel/create_distro_tarball.sh [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+# It generates the tarball including the configure (or rather generate.sh)
+# script.
+
+set -e  # exit script if any command returns an error
+set -u  # exit the script if any variable is uninitialized
+
+function usage {
+  echo -n "create_distro_tarball.sh [ --checkout_dir dir ] tarball"
+  echo " [ --git_tag tag | source_tree ]"
+  echo "examples of git_tag  would be 'master' or '1.11.33.4'"
```
Comment (jeffkaufman) [inline devel/create_distro_tarball.sh:31]: Can you clarify that `git_tag` can be a branch, tag, or commit id?  Basically, anything you can pass to `checkout -b`.

File: devel/create_distro_tarball.sh [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+CHECKOUT_DIR=
+if [ "$1" = "--checkout_dir" ]; then
+  if [ -z $2 ]; then
+    usage
+  fi
+  CHECKOUT_DIR=$2
+  shift 2
+else
+  CHECKOUT_DIR=`mktemp -d`
```
Comment (jeffkaufman) [inline devel/create_distro_tarball.sh:74]: can we swap the backtick command invocations for `$(...)`, here and below?

File: devel/create_distro_tarball.sh [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+# so we just compute it now, and save it there.
+cd src/
+./build/lastchange.sh $PWD > LASTCHANGE.in
+cd ..
+
+# It's tarball time!
+# TODO(morlovich): Fix giflib.gyp to support proper external depends.
+# Note that this is highly-version specific, and may require tweaks for
+# every release as its dependencies change. It's probably a good idea to
+# branch this script with every release.
```
Comment (jeffkaufman) [inline devel/create_distro_tarball.sh:133]: Can you change this to be more explicit?  Something like:

```
# Note that this is highly-version specific, and requires tweaks for every
# release as its dependencies change.  Always run the version of this
# script that's on the branch you're making a tarball for.
```

File: devel/create_distro_tarball.sh [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+tar cj --dereference --exclude='.git' --exclude='.svn' --exclude='.hg' -f $TARBALL \
```
Comment (jeffkaufman) [inline devel/create_distro_tarball.sh:134]: `--exclude='.svn' --exclude='.hg'` shouldn't be needed anymore now that all our dependencies are git.

File: devel/create_distro_tarball.sh [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    $DIR/src/third_party/modp_b64 \
+    $DIR/src/third_party/optipng \
+    $DIR/src/third_party/protobuf \
+    $DIR/src/third_party/rdestl \
+    $DIR/src/third_party/re2 \
+    $DIR/src/third_party/redis-crc \
+    $DIR/src/third_party/serf \
+    $DIR/src/third_party/zlib/zlib.gyp \
+    $DIR/src/tools/gyp \
+    $DIR/src/url
```
Comment (jeffkaufman) [inline devel/create_distro_tarball.sh:183]: Could we exclude the windows dlls that lintian was objecting to here as well?

**Review by jeffkaufman — CHANGES_REQUESTED** [review]

**Comment by morlovich** [comment]
I am going to re-do it starting with a merge off branch 33 one, I think.

**Pull request closed without merging**